napari_allencell_annotator/widgets/files_widget.py

```diff
@@ -47,14 +49,41 @@ def __init__(self, annotator_model: AnnotatorModel):
         self.setCurrentItem(None)
         self._annotator_model = annotator_model
 
-        self._annotator_model.image_changed.connect(
-            lambda: self.setCurrentItem(self.item(self._annotator_model.get_curr_img_index()))
-        )
+        self._annotator_model.image_changed.connect(self._handle_image_changed)
         self._annotator_model.images_shuffled.connect(self._handle_shuffle)
         self._annotator_model.image_set_added.connect(
             lambda: self._handle_shuffle(self._annotator_model.is_images_shuffled())
         )
         self._annotator_model.image_count_changed.connect(self._handle_image_count_change)
+        self.currentItemChanged.connect(self._handle_file_item_changed)
+
+    def _handle_image_changed(self):
+        with QSignalBlocker(self):
+            if self._annotator_model.get_curr_img_index() is not None:
+                self.setCurrentItem(self.item(self._annotator_model.get_curr_img_index()))
+            else:
+                self.setCurrentItem(None)
+
+    def _handle_file_item_changed(self, curr_item: FileItem, prev_item: FileItem) -> None:
+        """
+        Update the model when the current item in the file widget is changed if it has not been updated.
+
+        Parameters
+        ----------
+        curr_item:
+            the current file item
+        prev_item:
+            the previous file item
+        """
+        if prev_item is None:
```

Comment:
with the `-1` changes we no longer need `None` checks here right?

Reply:
Yes. I'm not sure if I should change it here or in #68 ?